src/components/ChatSection/SendMessagePanel/SendMessagePanelContainer.js

```diff
@@ -1,51 +1,46 @@
 import React from 'react';
 import SendMessagePanel from "./SendMessagePanel";
-import { connect } from "react-redux";
-import { addNewMessage } from "../../../redux/dialogsData/dialogsDataActions";
 import { v4 as uuidv4 } from "uuid";
 import moment from "moment";
-import { compose } from "redux";
-import { withRouter } from "react-router-dom";
 
-const SendMessagePanelContainer = (props) => {
-    const findInterlocutor = () => {
-        const dialog = props.dialogsData.dialogs.find(dialog => dialog.dialogId === props.dialogsData.currentDialog)
-        return dialog ? dialog.members.find(m => m.id !== props.currentUser.id) : {}
+class SendMessagePanelContainer extends React.Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            dialog: null,
+            interlocutor: null
+        }
     }
 
-    const onSubmit = (formData) => {
-        props.addNewMessage(
-            uuidv4(),
-            props.dialogsData.currentDialog,
-            props.currentUser.id,
-            moment().format(),
-            formData.message
-        )
+    componentDidUpdate() {
+        if (!this.state.dialog || this.state.dialog.dialogId !== this.props.dialogsData.currentDialog) {
+            const dialog = this.props.dialogsData.dialogs.find(d => d.dialogId === this.props.dialogsData.currentDialog)
+            const interlocutor = dialog && this.findInterlocutor(dialog)
+            this.setState({ dialog, interlocutor })
+        }
     }
 
-    return (
-        <>
-            {
-                !props.dialogsData.currentDialog ? <div/> :
-                    <SendMessagePanel
-                        currentUser={ props.currentUser }
-                        interlocutor={ findInterlocutor() }
-                        currentDialog={ props.dialogsData.currentDialog }
-                        onChangeInput={ props.onChangeInput }
-                        onSubmit={ onSubmit }
-                    />
-            }
-        </>
-    )
-}
+    findInterlocutor = (dialog) => dialog.members.find(m => m.id !== this.props.currentUser.id)
```

Comment:
name of variable findInterlocutor should named in camelCase right?

Reply:
It is already camelCase